userdocs/diagnostics/trailing-closure-matching.md

```diff
@@ -0,0 +1,99 @@
+# Argument Matching for Trailing Closures
+
+In Swift, calling a function with one or more trailing closure arguments requires the label of the first trailing closure argument to be omitted. As a result, the compiler must consider additional context when determining which function parameter the trailing closure should satisfy.
+
+Before Swift 5.3, the compiler used a backward scanning rule to match a trailing closure to a function parameter. Starting from the end of the parameter list, it moved backwards until finding a parameter compatible with trailing closures. This could sometimes lead to unexpected behavior. Consider the following example:
```

Comment:
This explanation is a little vague, but I think that's OK since it describes the old behavior

Reply:
Maybe replace `compatible with trailing closures` with `which could accept a trailing closure argument i.e. function type, generic parameter, 'Any' type`?